<?xml version="1.0"?>
<launch>

    <arg name="trajectory_parameters"        default="$(find tracking)/cfg/trajectory_parameters.yaml" />
    <arg name="publish_visualization_marker" default="false" />
    <arg name="use_trajectory_generation"    default="true" />
    <arg name="mav_name"                     default="firefly" />
    <arg name="use_odometry_for_platform"    default="false" />
    
    <group if="$(arg use_trajectory_generation)" >
        <node pkg="tracking" type="tracking_trajectory_generation_tracker" name="tracking_trajectory_generation_tracker" output="screen" args="-publish_visualization_marker $(arg publish_visualization_marker) -use_odometry_for_platform $(arg use_odometry_for_platform)">
            <rosparam command="load" file="$(arg trajectory_parameters)" />
            <remap from="command_trajectory" to="$(arg mav_name)/command/track/trajectory" />
            <remap from="mav_odometry"       to="$(arg mav_name)/ground_truth/odometry" />
            <remap from="husky_odometry"     to="husky/odometry" />
            <remap from="model_states"       to="gazebo/model_states" />
        </node>
    </group>

    <group unless="$(arg use_trajectory_generation)" >
    	<node pkg="tracking" type="tracking_tracking_node" name="tracking_tracking_node" output="screen" args="-use_odometry_for_platform $(arg use_odometry_for_platform)">
            <remap from="command_pose" 	         to="$(arg mav_name)/command/track/pose" />
	        <remap from="quad_odometry"  	     to="$(arg mav_name)/ground_truth/odometry" />
	        <remap from="model_state" 		     to="gazebo/model_states" />
            <remap from="husky_odometry"         to="husky/odometry" />
        </node>
    </group>

    <group if="$(arg publish_visualization_marker)">
        <node pkg="rviz" type="rviz" name="rviz" args="-d $(find tracking)/rviz/trajectory_visualizer.rviz" />
    </group>

</launch>
